disk-buffering/src/main/java/io/opentelemetry/contrib/disk/buffering/exporters/ExporterCallback.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.disk.buffering.exporters;
+
+import io.opentelemetry.contrib.disk.buffering.SignalType;
+import javax.annotation.Nullable;
+
+/** Notifies about exporter and storage-related operations from within a signal to disk exporter. */
+public interface ExporterCallback {
+  /**
+   * Called when an export to disk operation succeeded.
+   *
+   * @param type The type of signal associated to the exporter.
+   */
+  void onExportSuccess(SignalType type);
+
+  /**
+   * Called when an export to disk operation failed.
+   *
+   * @param type The type of signal associated to the exporter.
+   * @param error Optional - provides more information of why the operation failed.
+   */
+  void onExportError(SignalType type, @Nullable Throwable error);
+
+  /**
+   * Called when the exporter is closed.
+   *
+   * @param type The type of signal associated to the exporter.
+   */
+  void onShutdown(SignalType type);
```

Comment:
Not a huge deal, but I often find it convenient to provide default empty methods for implementations that may not necessarily care about all of the interface methods. That way, implementers can choose to implement the ones that they are interested in, and it doesn't require many implementations to repeat empty bodies.

Reply:
IMO default method bodies should be avoided in interfaces that form public APIs. It makes discoverability harder, doesn't cleanly separate the interface from the implementation, and the behavior can become part of the public API's contract which can constrain changes in future.

Having said that, an empty body is less of a big deal to me than a body containing actual business logic would be

Reply:
I think @fractalwrench makes some good points here, but like I said, I have found them convenient. Just know that interfaces with multiple `onXXX` type methods (like this one) especially when on public APIs will have many implementers with no-op implementations. Check out `SpanProcessor` implementations, for example.

An alternative is to separate the methods into separate interfaces, but that's fairly intrusive to this existing design. It's not that important.